test/service_test.cc

```diff
@@ -111,6 +113,43 @@ class ServiceTest : public TestWithParam<Envoy::Network::Address::IpVersion> {
   int grpc_server_port_{0};
 };
 
+class ServiceTestWithParameterizedConstructor : public ServiceTest {
+public:
+  void SetUp() override {
+    Envoy::Thread::MutexBasicLockable log_lock_;
```

Comment:
Might want to make this a member instead of local variable to ensure the proper lifetime,
as the Logging::Context constructor takes a ref?:

```
  Context(spdlog::level::level_enum log_level, const std::string& log_format,
          Thread::BasicLockable& lock, bool should_escape, bool enable_fine_grain_logging = false);
```

Reply:
Agreed. I'm updating this PR to retest, since my local test can succeed somehow. Thanks Otto!